includes/wpml/class-lesson-progress.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * File containing \Sensei\WPML\Lesson_Progress class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\WPML;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Lesson_Progress
+ *
+ * Compatibility code with WPML.
+ *
+ * @since $$next-version$$
+ *
+ * @internal
+ */
+class Lesson_Progress {
+	/**
+	 * Init hooks.
+	 */
+	public function init() {
+		add_filter( 'sensei_utils_user_completed_lesson_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_create_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_get_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_has_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_delete_for_lesson_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_find_lesson_id', array( $this, 'translate_lesson_id' ), 10, 1 );
+	}
+
+	/**
+	 * Translate course ID.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param int $lesson_id Lesson ID.
+	 * @return int
+	 */
+	public function translate_lesson_id( $lesson_id ) {
+		$details = (array) apply_filters(
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			'wpml_element_language_details',
+			null,
+			array(
+				'element_id'   => $lesson_id,
+				'element_type' => 'lesson',
+			)
+		);
+
+		$original_language_code = $details['source_language_code'] ?? $details['language_code'] ?? null;
+
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+		return (int) apply_filters( 'wpml_object_id', $lesson_id, 'lesson', true, $original_language_code );
+	}
```

Comment:
Nitpick, but all these translate functions look very similar except for a couple of different values in the apply_filter line, should we move this to a common place instead? Maybe all these entity specific translation classes can have a parent class that contains this function or an util class or something? WDYT?

Reply:
There is a separate PR where we extract all WPML filters and actions in a form of methods into a trait: https://github.com/Automattic/sensei/pull/7566

Without WPML details these methods look much shorter and clearer, I don't think it makes sense to create additional abstraction here:
https://github.com/Automattic/sensei/blob/1fe078cf18c7be1d8d48d661d9343d14307483e2/includes/wpml/class-lesson-progress.php#L48-L54